core/src/main/java/jenkins/model/logging/impl/StreamLogStorage.java

```diff
@@ -0,0 +1,104 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.model.logging.impl;
+
+import hudson.console.ConsoleLogFilter;
+import hudson.model.AbstractBuild;
+import hudson.model.BuildableItemWithBuildWrappers;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.model.StreamBuildListener;
+import hudson.tasks.BuildWrapper;
+import jenkins.model.logging.Loggable;
+import jenkins.model.logging.LogStorage;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.Beta;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Logging method which takes {@link OutputStream} as a destination.
+ * This implementation consults with {@link ConsoleLogFilter} extensions in Jenkins.
+ * @author Oleg Nenashev
+ * @since TODO
+ */
+@Restricted(Beta.class)
+public abstract class StreamLogStorage extends LogStorage {
+
+    private static final Logger LOGGER =
+            Logger.getLogger(StreamLogStorage.class.getName());
+
+    public StreamLogStorage(@Nonnull Loggable loggable) {
+        super(loggable);
+    }
+
+    public abstract OutputStream createOutputStream() throws IOException, InterruptedException;
+
+    /**
+     * Defines an additional Console Log Filter to be used with the logging method.
+     * This filter may be also used for log redirection and multi-reporting in very custom cases.
+     * @return Log filter. {@code null} if no custom implementation
+     */
+    public ConsoleLogFilter getExtraConsoleLogFilter() {
+        return null;
+    }
+
+    //TODO: This decoration logic should be shared with other implementations
+    @Override
+    public final StreamBuildListener createBuildListener() throws IOException, InterruptedException {
+
+        OutputStream logger = createOutputStream();
+        if (!(loggable instanceof Run<?,?>)) {
+            throw new IOException("Loggable is not a Run instance: " + loggable.getClass());
+        }
+        Run<?,?> build = (Run<?, ?>)loggable;
+
+        // Global log filter
+        for (ConsoleLogFilter filter : ConsoleLogFilter.all()) {
+            logger = filter.decorateLogger(build, logger);
+        }
+        final Job<?, ?> project = build.getParent();
+
+        // Project specific log filters
+        if (project instanceof BuildableItemWithBuildWrappers && build instanceof AbstractBuild) {
```

Comment:
…and then override that in `AbstractBuildExecution`, cleaning this mistake up at last.

`WorkflowRun` does not yet support `ConsoleLogFilter.all()` but that can be dealt with separately: https://github.com/jenkinsci/workflow-job-plugin/pull/27/files#diff-68f690dbd3fa2d9fb89d6af3f8fd7216R218